### PR TITLE
Move SSO checking into Lock init

### LINKED ISSUE #414: Better behavior when logged-in users visits wp-login.php
Basic WordPress behavior when viewing the login page is to show the login form again but when Auth0 is installed, you get the WP logo and a link:

screenshot 2018-03-22 12 38 52

I think redirecting to the default login URL would be more helpful here. Also, if there is a `redirect_to` parameter, we should use that.

### LINKED ISSUE #508: Single sign on shows the login username/password fields briefly before automatically signing in
Steps:
- Plugin configured in Wordpress with SSO enabled.
- Logged into auth0 account from the same tenant (different application) using the hosted login page.
- Link is clicked that goes to a Wordpress page that requires login and automatically redirects to the login page.

Expected behavior:
- User is taken to the login page.
- Lock widget is loaded and has the "spinner" animation.
- User is authenticated and redirected to the originally requested page (via redirect URL)

Actual behavior:
- User is taken to the login page.
- Lock widget is loaded and shows "spinner" briefly.
- **Spinner animation is removed and the username/password fields are displayed in the lock widget.**
- The page continues working and after a few more seconds redirects the user to the originally requested page.

As you can see, while the SSO works there is some confusion as it looks like it is requiring the user to enter credentials before it completes the SSO. This is mostly a cosmetic thing but may confuse users depending on how long it sits on the page with the login fields present.

### PR DESCRIPTION
**Pull request by joshcanhelp**
### Changes

Moving the SSO processing from a separately-included HTML template into the Lock init JS file. 

- Remove `templates/auth0-sso-handler-lock10.php` file
- Move inline JS from the above to `assets/js/lock-init.js` to hide the Lock form until the SSO check is complete
- Redirect users from the `wp-login.php` page if they are already logged in

### References

Closes #508 
Closes #414 

### Testing

Testing steps:

1. Turn on SSO in the Auth0 settings page
2. Clear out all WordPress auth cookies
3. Visit the `wp-login.php` page

**Expected**

No Lock form shows and you're redirected to the default login link with a WP session and no JS console errors. 

1. Visit the `wp-login.php` page again

You're redirected to the default login link with the WP session intact.

* [x] This change adds unit test coverage
* [x] This change has been tested on the latest version of the platform

### Checklist

* [x] All existing and new tests complete without errors
* [x] All code quality tools/guidelines in the [Contribution guide](CONTRIBUTION.md) have been run/followed
* [x] All relevant assets have been compiled as directed in the [Contribution guide](CONTRIBUTION.md), if applicable
* [x] All active GitHub CI checks have passed

### CODE REVIEW

File: lib/WP_Auth0_Lock10_Options.php [hunk 1]
```diff
@@ -134,13 +134,8 @@ protected function build_settings( $settings ) {
 
 	public function get_sso_options() {
 		$options['scope'] = WP_Auth0_LoginManager::get_userinfo_scope( 'sso' );
-
```
Comment (joshcanhelp) [inline lib/WP_Auth0_Lock10_Options.php]: This is only ever used for SSO, which uses the hybrid auth (modified implicit) so there's never a need for a code callback processing.

File: lib/WP_Auth0_LoginManager.php [hunk 2]
```diff
@@ -90,11 +90,9 @@ public function init() {
 		add_action( 'login_init', array( $this, 'login_auto' ) );
 		add_action( 'template_redirect', array( $this, 'init_auth0' ), 1 );
 		add_action( 'wp_logout', array( $this, 'logout' ) );
-		add_filter( 'login_message', array( $this, 'auth0_sso_footer' ) );
 		add_action( 'wp_footer', array( $this, 'auth0_singlelogout_footer' ) );
 		add_action( 'admin_footer', array( $this, 'auth0_singlelogout_footer' ) );
 		add_action( 'login_footer', array( $this, 'auth0_singlelogout_footer' ) );
-		add_action( 'wp_login', array( $this, 'end_session' ) );
```
Comment (joshcanhelp) [inline lib/WP_Auth0_LoginManager.php]: Deprecated in #557 but not removed from the login page.

File: lib/WP_Auth0_LoginManager.php [hunk 3]
```diff
@@ -90,11 +90,9 @@ public function init() {
 		add_action( 'login_init', array( $this, 'login_auto' ) );
 		add_action( 'template_redirect', array( $this, 'init_auth0' ), 1 );
 		add_action( 'wp_logout', array( $this, 'logout' ) );
-		add_filter( 'login_message', array( $this, 'auth0_sso_footer' ) );
```
Comment (joshcanhelp) [inline lib/WP_Auth0_LoginManager.php]: Removes the existing SSO template `templates/auth0-sso-handler-lock10.php`

File: WP_Auth0.php [hunk 4]
```diff
@@ -372,6 +372,18 @@ public function render_form( $html ) {
 			return $html;
 		}
 
+		// If the user has a WP session, determine where they should end up and redirect.
+		if ( is_user_logged_in() ) {
+			$login_redirect = ! empty( $_REQUEST['redirect_to'] ) ?
```
Comment (cocojoe) [inline WP_Auth0.php]: I think in logic usually best to avoid extra operands, you don't need the `!` if you switch the Ternary values.

**Comment by joshcanhelp** [comment]
@cocojoe - Fixed up, tinkered with `WP_Auth0` a bit to assist with testing, and added tests 👍

**Comment by joshcanhelp** [comment]
@cocojoe - Bump is kind of fake as I removed a few external libraries from the test calculation ... don't tell anyone!

**Pull request merged**